<!-- -->
<launch>
	<arg name="robot_namespace"     default="drone1" />
	<arg name="odometry_topic" default="odometry/filtered"/>
	<arg name="localization_topic" default="self_localization/pose"/>
	<arg name="speed_topic" default="self_localization/speed"/>
	<param name="robot_namespace" value="$(arg robot_namespace)"/>
	<node pkg="tf2_ros" type="static_transform_publisher" name="gps_tf" args="0 0 0 0 0 0 map base_link" respawn="true"/>
	<node pkg="tf2_ros" type="static_transform_publisher" name="gps_tf2" args="0 0 0 0 0 0 odom map" respawn="true"/>
	<!--<node pkg="tf2_ros" type="static_transform_publisher" name="gps_tf3" args="1 1 1 1 1 1 map gps" respawn="true"/>-->

	<group ns="$(arg robot_namespace)">
		<param name="robot_namespace" value="$(arg robot_namespace)"/>
		<node pkg="robot_localization" type="navsat_transform_node" name="navsat_transform_node" respawn="true">
			<rosparam param="publish_tf">"true"</rosparam>
			<rosparam param="frequency">"10"</rosparam>
			<rosparam param="magnetic_declination_radians">"0"</rosparam>
			<rosparam param="yaw_offset">"0"</rosparam>
			<rosparam param="wait_for_datum">"false"</rosparam>
			<rosparam param="broadcast_utm_transform">"true"</rosparam>
			<rosparam param="broadcast_utm_transform_as_parent_frame">"true"</rosparam>
			<rosparam param="publish_filtered_gps">"true"</rosparam>
			<rosparam param="use_odometry_yaw">"false"</rosparam>
			<remap from="imu/data" to="mavros/imu/data" />
			<remap from="gps/fix" to="mavros/global_position/raw/fix" />
		</node>

		<node pkg="robot_localization" type="ekf_localization_node" name="ekf_localization_node" respawn="true">
			<rosparam param="publish_tf">"true"</rosparam>
			<rosparam param="sensor_timeout">"0.2"</rosparam>
			<rosparam param="two_d_mode">"false"</rosparam>
			<rosparam param="map_frame">"map"</rosparam>
			<rosparam param="odom_frame">"odom"</rosparam>
			<rosparam param="base_link_frame">"base_link"</rosparam>
			<rosparam param="world_frame">"odom"</rosparam>
			<rosparam param="odom0">"odometry/gps"</rosparam>
			<rosparam param="odom0_config">[true,  true,  true, false, false, false, false, false, false, false, false, false, false, false, false]</rosparam>
			<rosparam param="odom0_differential">"false"</rosparam>
			<param name="imu0" value="mavros/imu/data"/>
			<rosparam param="imu0_config">[false,  false,  false, true, true, true, false, false, false, true, true, true, false, false, false]</rosparam>
		</node>

		<param name="odometry_topic" type="str" value="$(arg odometry_topic)"/>
		<param name="localization_topic" type="str" value="$(arg localization_topic)"/>
		<param name="speed_topic" type="str" value="$(arg speed_topic)"/>
		<node pkg="gps_localization" type="gps_localization_node" name="gps_localization_node" respawn="true">
	</node>
	</group>
	<node pkg="topic_tools" type="relay" name="speed_relay" args="$(arg robot_namespace)/mavros/local_position/velocity_local $(arg robot_namespace)/self_localization/speed" respawn="true"></node>
</launch>
